### PR TITLE
Add the GRATIA troubleshooting document.

### PR DESCRIPTION
**Pull request by djw8605**
SOFTWARE-3065

The *.graffle is the save format of omnigraffle.  I included my image source!

### CODE REVIEW

File: docs/other/troubleshooting-gratia.md [hunk 1]
```diff
@@ -0,0 +1,375 @@
...
+4.  Type a friendly email that asks the GRACC team to change your site name at the collector. Make sure to tell them the old name and the new name.
+
+ Is a site reporting data?
+--------------------------
+
+You can see if the OSG GRACC Server is getting data from a site by going to [GRACC](https://gracc.opensciencegrid.org/dashboard/db/pilot-jobs-summary?orgId=1):
+
+1.  Specify the site name in Facility
+
+Condor must be configured to put information about each job into a special directory.
```
Comment (brianhlin) [inline docs/other/troubleshooting-gratia.md]: Holy section title, batman. Maybe "Verifying Condor Job History Location"?

Also, should there be a note that this is for condor batch systems only?

File: docs/other/troubleshooting-gratia.md [hunk 2]
```diff
@@ -0,0 +1,375 @@
...
+-------------------------------------------------------------------------------------
+
+Gratia will read and remove the files in order to collect the accounting information.
+
+The configuration variable is called `PER_JOB_HISTORY_DIR`. If you install the OSG RPM for Condor, the Gratia probe will extend its configuration by adding a file to `/etc/condor/config.d`, and will set this variable to `/var/lib/gratia/data`. If you are using a different installation method, you probably need to set the variable yourself. You can check if it's set by using `condor_config_val`, like this:
```
Comment (brianhlin) [inline docs/other/troubleshooting-gratia.md]: probably -> may

File: docs/other/troubleshooting-gratia.md [hunk 3]
```diff
@@ -0,0 +1,375 @@
...
+
+If you set this value, you need to restart condor:
+
+    :::console
+    root@host # condor_restart
+    Sent "Restart" command to local master
+
+Unlike many Condor settings, a **condor\_reconfig** is not sufficient - you must restart!
+
+If you accidentally did not set `PER_JOB_HISTORY_DIR` (see above)
```
Comment (brianhlin) [inline docs/other/troubleshooting-gratia.md]: Make this a level three heading and call it `Condor history recovery` or something

File: docs/other/troubleshooting-gratia.md [hunk 4]
```diff
@@ -0,0 +1,375 @@
...
+    13:35:29 CDT Gratia: Removing log files older than 31 days from /var/log/gratia
+    13:35:29 CDT Gratia: /var/log/gratia uses 0.035% and there is 73% free
+    13:35:29 CDT Gratia: Removing incomplete data files older than 31 days from /var/lib/gratia/data/
+    13:35:29 CDT Gratia: /var/lib/gratia/data uses 0% and there is 73% free
+    13:35:29 CDT Gratia: End of execution summary: new records sent successfully: 37
+
+
+
+!!! note 
+    Condor rotates history files, so you can only report what Condor has kept. Controlling the Condor history is documented in the Condor manual. In particular, see the options for [MAX\_HISTORY\_LOG](http://research.cs.wisc.edu/condor/manual/v7.6/3_3Configuration.html#param:MaxHistoryLog) and [MAX\_HISTORY\_ROTATIONS](http://research.cs.wisc.edu/condor/manual/v7.6/3_3Configuration.html#param:MaxHistoryRotations).
```
Comment (brianhlin) [inline docs/other/troubleshooting-gratia.md]: Let's update the link to a modern version of condor

File: docs/other/troubleshooting-gratia.md [hunk 5]
```diff
@@ -0,0 +1,375 @@
...
+    %RED%Later, Gratia failed to connect to the server due to a bad hostname%ENDCOLOR%
+    15:06:55 CDT Gratia: Failed to send xml to web service due to an error of type "socket.gaierror": (-2, 'Name or service not known')
+    ...
+    15:06:55 CDT Gratia: Response indicates failure, the following files will not be deleted:
+    15:06:55 CDT Gratia:    /var/lib/gratia/tmp/gratiafiles/
+        subdir.condor_fermicloud084.fnal.gov_ggratia-osg-itb.opensciencegrid.org_80/
+        outbox/r.30604.condor_fermicloud084.fnal.gov_ggratia-osg-itb.opensciencegrid.org_80.gratia.xml__wfIgi30606
+
+
+If you accidentally had a bad Gratia hostname, you probably want to recover your Gratia data.
```
Comment (brianhlin) [inline docs/other/troubleshooting-gratia.md]: Level 3 header, `Bad hostname recovery`

File: docs/other/troubleshooting-gratia.md [hunk 6]
```diff
@@ -0,0 +1,375 @@
...
+        root@host # cd !$
+
+        %RED%# For each tz file:%ENDCOLOR%
+        root@host # tar xf tz.1223.... [name shortened for legibility]
+        root@host # rm tz.1223....
+
+
+    When you've done this, you can re-run the Gratia probe by hand, or wait for it to run via cron.
+
+Appendix: Important Gratia files
```
Comment (brianhlin) [inline docs/other/troubleshooting-gratia.md]: Rename this "Reference"

File: docs/other/troubleshooting-gratia.md [hunk 7]
```diff
@@ -0,0 +1,375 @@
...
+| `/var/log/gratia/gridftpTransfer.log`               | Log file specific to the Gratia GridFTP probe                                                                                                                                      |
+| `/var/lib/gratia/data`                              | Location for Condor and PBS job data before being processed by Gratia<br>Condor's `PER_JOB_HISTORY_DIR` should be set to this location                                       |
+| `/var/lib/gratia/tmp/gratiafiles`                   | Location for temporary Gratia data as it is being processed, usually empty.<br>If you have files that are more than 30 minutes old in this directory, there may be a problem |
+| `/etc/gratia/%RED%PROBE-NAME%ENDCOLOR%/ProbeConfig` | Configuration for Gratia probes, one per probe type</br>Normally you don't need to edit this                                                                                 |
+
+The most common RPMs you will see are:
```
Comment (brianhlin) [inline docs/other/troubleshooting-gratia.md]: We should indicate that you will not see all of these on a single host

File: docs/other/troubleshooting-gratia.md [hunk 8]
```diff
@@ -0,0 +1,375 @@
...
+        root@host # tar xf tz.1223.... [name shortened for legibility]
+        root@host # rm tz.1223....
+
+
+    When you've done this, you can re-run the Gratia probe by hand, or wait for it to run via cron.
+
+Appendix: Important Gratia files
+--------------------------------
+
+This document cannot cover all the errors you might experience. If you need to look for more data, you can look at log files for the various services on your CE.
```
Comment (brianhlin) [inline docs/other/troubleshooting-gratia.md]: The first sentence is weird, get rid of it

File: docs/other/troubleshooting-gratia.md (hunk 4 above)
Comment (djw8605) [reply docs/other/troubleshooting-gratia.md]: Is there a way to link directly to a parameters?

File: docs/other/troubleshooting-gratia.md (hunk 7 above)
Comment (djw8605) [reply docs/other/troubleshooting-gratia.md]: Eh, most of them will be.  The only entry that will not be on every host clearly says "Log file specific to the Gratia GridFTP"

File: docs/other/troubleshooting-gratia.md (hunk 4 above)
Comment (brianhlin) [reply docs/other/troubleshooting-gratia.md]: I normally search for the variable in the index and find the relevant link for the configuration section (3.5 for 8.6). Here are the links:

- http://research.cs.wisc.edu/htcondor/manual/v8.6/3_5Configuration_Macros.html#21780
- http://research.cs.wisc.edu/htcondor/manual/v8.6/3_5Configuration_Macros.html#21785

File: docs/other/troubleshooting-gratia.md [hunk 9]
```diff
@@ -0,0 +1,381 @@
...
+| `/var/lib/gratia/tmp/gratiafiles`                   | Location for temporary Gratia data as it is being processed, usually empty.<br>If you have files that are more than 30 minutes old in this directory, there may be a problem |
+| `/etc/gratia/%RED%PROBE-NAME%ENDCOLOR%/ProbeConfig` | Configuration for Gratia probes, one per probe type</br>Normally you don't need to edit this                                                                                 |
+
+The most common RPMs you will see are:
+
+| RPM                             | Purpose                                                                             |
+|:--------------------------------|:------------------------------------------------------------------------------------|
+| `gratia-probe-common`           | Code shared between all Graita probes                                               |                            |
+| `gratia-probe-condor`           | The probe that tracks Condor usage                                         |
+| `gratia-probe-slurm`           | The probe that tracks Condor usage                                         |
```
Comment (brianhlin) [inline docs/other/troubleshooting-gratia.md]: This should say it tracks slurm usage

File: docs/other/troubleshooting-gratia.md (hunk 7 above)
Comment (brianhlin) [reply docs/other/troubleshooting-gratia.md]: I believe that I was referring to the RPMs below

File: docs/other/troubleshooting-gratia.md (hunk 4 above)
Comment (djw8605) [reply docs/other/troubleshooting-gratia.md]: Do you crack open the raw html to get these numbers?  Do these numbers change for future releases?

Comment (brianhlin) [reply docs/other/troubleshooting-gratia.md]: Nope, I'm not that sophisticated. Here's my process:

1. Go [here](http://research.cs.wisc.edu/htcondor/manual/v8.6/Index.html)
1. Search for the variable in question
1. Right-click and copy the link location for the 3.5.1 link

I believe the numbers do change between releases. @timtheisen may have details on how that works.

Comment (djw8605) [reply docs/other/troubleshooting-gratia.md]: Ah, ok.  That index, didn't get that.  Makes sense, Thanks.

**Comment by djw8605** [comment]
@brianhlin Travis is failing due to a link in gridftp page?

**Comment by brianhlin** [comment]
Yup, I'm seeing that fail in other PRs, too. Overriding CI and merging.

**Pull request merged**